pkg/pillar/cmd/zedrouter/acl.go

```diff
@@ -770,141 +804,269 @@ func diffIpsets(newIpsets, oldIpsets []string) ([]string, []string, bool) {
+			action0.PortMap != action1.PortMap {
+			return false
 		}
-		args = append(args, rule...)
-		if ipVer == 4 {
-			err = iptables.IptableCmd(args...)
-		} else if ipVer == 6 {
-			err = iptables.Ip6tableCmd(args...)
-		} else {
-			err = errors.New(fmt.Sprintf("ACL: Unknown IP version %d", ipVer))
+		if action0.PortMap {
+			if action0.TargetPort != action1.TargetPort {
+				return false
+			}
 		}
-		if err != nil {
-			return err
+		if action0.Limit {
+			if action0.LimitRate != action1.LimitRate ||
+				action0.LimitUnit != action1.LimitUnit ||
+				action0.LimitBurst != action1.LimitBurst {
+				return false
+			}
 		}
 	}
-	return nil
+	return true
 }
 
-func deleteACLConfiglet(bridgeName string, vifName string, isMgmt bool,
-	ACLs []types.ACE, bridgeIP string, appIP string) error {
-
-	log.Infof("deleteACLConfiglet: ifname %s vifName %s ACLs %v\n",
-		bridgeName, vifName, ACLs)
-
-	ipVer := determineIpVer(isMgmt, bridgeIP)
-	rules, err := aclToRules(bridgeName, vifName, ACLs, ipVer,
-		bridgeIP, appIP)
-	if err != nil {
-		return err
+// check for duplicate portmap rules in same set of ACLs
+// for this, we will match either the target port or
+// the ingress protocol/lport being same
+func matchACLForPortMap(ACLs []types.ACE) bool {
+	matchTypes := []string{"protocol", "lport"}
+	idx := 0
+	ruleNum := len(ACLs)
+	for idx < ruleNum-1 {
+		ace := ACLs[idx]
+		for _, action := range ace.Actions {
+			if !action.PortMap {
+				continue
+			}
+			idx1 := idx + 1
+			for idx1 < ruleNum {
+				ace1 := ACLs[idx1]
+				for _, action1 := range ace1.Actions {
+					if !action1.PortMap {
+						continue
+					}
+					if action.TargetPort == action1.TargetPort ||
+						checkForMatchCondition(ace, ace1, matchTypes) {
+						return true
+					}
+				}
+				idx1++
+			}
+		}
+		idx++
 	}
-	dropRules, err := aclDropRules(bridgeName, vifName)
-	if err != nil {
-		return err
+	return false
+}
+
+// check for duplicate portmap rules in between two set of ACLs
+// for this, we will match the ingress protocol/lport being same
+func matchACLsForPortMap(ACLs []types.ACE, ACLs1 []types.ACE) bool {
+	matchTypes := []string{"protocol", "lport"}
+	for _, ace := range ACLs {
+		for _, action := range ace.Actions {
+			// not a portmap rule
+			if !action.PortMap {
+				continue
+			}
+			for _, ace1 := range ACLs1 {
+				for _, action1 := range ace1.Actions {
+					// not a portmap rule
+					if !action1.PortMap {
+						continue
+					}
+					// check for ingress protocol/port
+					if checkForMatchCondition(ace, ace1, matchTypes) {
+						return true
+					}
+				}
+			}
+		}
 	}
-	rules = append(rules, dropRules...)
-	for _, rule := range rules {
-		log.Debugf("deleteACLConfiglet: rule %v\n", rule)
-		args := rulePrefix("-D", isMgmt, ipVer, vifName, appIP, rule)
-		if args == nil {
-			log.Debugf("deleteACLConfiglet: skipping rule %v\n",
-				rule)
-			continue
+	return false
+}
+
+// generic comparision routine for ACL match conditions
```

Comment:
Per the comments above you should only match protocol and fport.
For instance, if the two ACLs have two different ip or host matches, we should still declare them to be duplicate/conflicting.

Reply:
Currently, I see, the lport is getting filled up by cloud. [{"Name":"eth0","AppMacAddr":null,"AppIPAddr":"","Error":"","Network":"0edb5a7f-44d7-42a6-a247-c490f0c91fe8","ACLs":[{"Matches":[{"Type":"protocol","Value":"tcp"},{"Type":"lport","Value":"8022"}],"Actions":[{"Drop":false,"Limit":false,"LimitRate":0,"LimitUnit":"","LimitBurst":0,"PortMap":true,"TargetPort":22}]},{"Matches":[{"Type":"host","Value":"*"}],"Actions":[{"Drop":false,"Limit":true,"LimitRate":10,"LimitUnit":"m","LimitBurst":40,"PortMap":false,"TargetPort":0}]},{"Matches":[{"Type":"ip","Value":"0.0.0.0/0"}],"Actions":[{"Drop":false,"Limit":true,"LimitRate":10,"LimitUnit":"m","LimitBurst":40,"PortMap":false,"TargetPort":0}]}]}],
Is this ok to add check for both for lport/fport.

Reply:
Sorry, my mistake. lport is correct.